subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/maven/MavenPomRelocationIntegrationTest.groovy

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.maven
+
+import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
+import spock.lang.Issue
+import spock.lang.Unroll
+
+class MavenPomRelocationIntegrationTest extends AbstractHttpDependencyResolutionTest {
+
+    @Issue('https://github.com/gradle/gradle/issues/1789')
+    @Unroll
+    def "succeed when relocation exists"() {
+        given:
+        def original = publishPomWithRelocation('groupA', 'projectA', relocationGroupId, relocationArtifactId)
+        def newModule = mavenHttpRepo.module(newGroupId, newArtifactId, "1.0").publish()
+
+        and:
+        createBuildFileWithDependency('groupA', 'projectA')
+
+        and:
+        original.pom.expectGet()
+        newModule.pom.expectGet()
+        newModule.artifact.expectGet()
+
+        when:
+        run "retrieve"
+
+        then:
+        file("libs").assertHasDescendants("${newArtifactId}-1.0.jar")
+
+        where:
+        relocationGroupId | relocationArtifactId | newGroupId  | newArtifactId
+        'groupB'          | 'projectB'           | 'groupB'    | 'projectB'
+        'newGroupA'       | ''                   | 'newGroupA' | 'projectA'
+    }
+
+    def "double-relocation should succeed"() {
+        given:
+        def moduleA = publishPomWithRelocation('groupA', 'projectA', 'groupB', 'projectB')
+        def moduleB = publishPomWithRelocation('groupB', 'projectB', 'groupC', 'projectC')
+        def moduleC = mavenHttpRepo.module('groupC', 'projectC', "1.0").publish()
+
+        and:
+        createBuildFileWithDependency('groupA', 'projectA')
+
+        and:
+        moduleA.pom.expectGet()
+        moduleB.pom.expectGet()
+        moduleC.pom.expectGet()
+        moduleC.artifact.expectGet()
+
+        when:
+        run "retrieve"
+
+        then:
+        file("libs").assertHasDescendants("projectC-1.0.jar")
+    }
+
+    def "fail when artifact in <relocation> doesn't exist"() {
```

Comment:
Maybe be more expressive about the test case e.g. "fails to resolve module if published artifact does not exist with relocated coordinates".